### PR TITLE
[Maps] implement validation for map question settings

### LINKED ISSUE #11199: [Maps] Validation for editing a map question
Add the server side validation rules for creating or editing filters for a map question.

The rules are:
1. Keys is selected for Location, Address, and Details URL, and they match the possible keys from the GeoJSON stored in the database.
2. There are no more than 6 filters.
3. Every filter has a key selected and a display name.

### PR DESCRIPTION
**Pull request by tallulahkay**
### Description

Implement validation for map question settings.

Rules:
1. Every field is required.
2. No more than 6 filters.

Used claude to come up with test cases

## Release notes

Continued development of map question.

### Instructions for manual testing

1. Set `map_question_enabled=true` in `application.dev.conf`
4. Create a map question (can use one of Seattle's public endpoint to test: `https://earlylearning.powerappsportals.us/provider-list-ccap/`)
5. Confirm that validation is added for making all question settings fields required, with max 6 filters.

### Issue(s) this completes

Fixes #11199

### CODE REVIEW

File: server/app/services/question/types/MapQuestionDefinition.java [hunk 1]
```diff
@@ -37,12 +39,58 @@ ImmutableSet<CiviFormError> internalValidate(Optional<QuestionDefinition> previo
...
+    boolean hasAddressKey = false;
+    boolean hasDetailsUrlKey = false;
+    int filterCount = 0;
+
+    for (QuestionSetting setting : settings) {
+      boolean isValidSetting = setting.settingValue() != null && !setting.settingValue().isEmpty();
+
+      switch ((MapSettingType) setting.settingType()) {
+        case LOCATION_NAME_GEO_JSON_KEY -> {
+          if (isValidSetting) hasNameKey = true;
```
Comment (melanie-exygy) [inline server/app/services/question/types/MapQuestionDefinition.java:60]: **suggestion (non-blocking):** Instead of creating a variable and setting it to true here, I think we could just do:
`if (!isValidSetting) { errors.add(CiviFormError.of("Name key cannot be empty"));`.  That seems more straightforward to me.

File: server/app/services/question/types/MapQuestionDefinition.java [hunk 2]
```diff
@@ -37,12 +39,58 @@ ImmutableSet<CiviFormError> internalValidate(Optional<QuestionDefinition> previo
 
     if (maxLocationSelections.isPresent()) {
       if (maxLocationSelections.getAsInt() < 1) {
-        errors.add(CiviFormError.of("Max location selections cannot be less than 1"));
+        errors.add(CiviFormError.of("Maximum location selections cannot be less than 1"));
       }
+    } else {
+      errors.add(CiviFormError.of("Maximum location selections is required"));
```
Comment (melanie-exygy) [inline server/app/services/question/types/MapQuestionDefinition.java]: **suggestion (non-blocking):** For consistency, could we make this similar to the error messages below and say "Maximum location selections cannot be empty"?

File: browser-test/src/admin/admin_map_question.test.ts [hunk 3]
```diff
@@ -69,10 +69,8 @@ test.describe('Create and edit map question', () => {
...
 
-        const filterKeySelect = page.getByLabel('Key', {exact: true})
-        const filterDisplayInput = page.getByLabel('Display name', {
-          exact: true,
-        })
+        const filterKeySelect = page.getByTestId('key-select')
+        const filterDisplayInput = page.getByTestId('display-name-input')
 
         await expect(filterKeySelect).toBeVisible()
         await expect(filterDisplayInput).toBeVisible()
```
Comment (melanie-exygy) [inline browser-test/src/admin/admin_map_question.test.ts:76]: **suggestion (blocking):** Could you add a test that checks that the filter button is disabled after 6 filters?  It also seems like we could use a validation test that checks that a toast appears if none of the settings are filled out.  Or maybe it just checks that you're still on the question edit page.

File: server/app/assets/stylesheets/styles.css [hunk 4]
```diff
@@ -44,6 +44,10 @@
     color: #005ea2;
   }
 
+  .add-map-filter.usa-button.usa-button--outline:disabled {
+    @apply bg-white text-gray-400 border border-gray-400;
```
Comment (melanie-exygy) [inline server/app/assets/stylesheets/styles.css:48]: **question (non-blocking):** Why do we need these custom styles for the disabled state?  Seems like the color contrast is pretty low this way.

File: server/test/services/question/types/MapQuestionDefinitionTest.java [hunk 5]
```diff
@@ -89,20 +91,22 @@ public void validate_withoutGeoJson_returnsError() {
             .build();
     QuestionDefinition question = new MapQuestionDefinition(config);
     assertThat(question.validate())
-        .containsOnly(CiviFormError.of("Map question must have valid GeoJSON"));
+        .contains(
+            CiviFormError.of("Map question must have valid GeoJSON"),
+            CiviFormError.of("Maximum location selections is required"));
```
Comment (melanie-exygy) [inline server/test/services/question/types/MapQuestionDefinitionTest.java]: **suggestion (non-blocking):** Since we're checking the max location selections too, could you add it to the test name?

File: server/app/assets/stylesheets/styles.css (hunk 4 above)
Comment (tallulahkay) [reply server/app/assets/stylesheets/styles.css:48]: This is to make it match the USWDS style because tailwind is still taking over for the admin side of things

File: server/app/services/question/types/MapQuestionDefinition.java (hunk 1 above)
Comment (tallulahkay) [reply server/app/services/question/types/MapQuestionDefinition.java:60]: We don't want to add the error twice in case there is more than one setting with LOCATION_NAME_GEO_JSON_KEY. There is nothing that creates a setting with LOCATION_NAME_GEOJSON_KEY more than once, but there's nothing enforcing that it only occurs once.

File: browser-test/src/admin/admin_map_question.test.ts (hunk 3 above)
Comment (tallulahkay) [reply browser-test/src/admin/admin_map_question.test.ts:76]: added both those tests!

File: server/app/services/question/types/MapQuestionDefinition.java (hunk 1 above)
Comment (melanie-exygy) [reply server/app/services/question/types/MapQuestionDefinition.java:60]: Ah, I see.  That makes sense.

**Review by melanie-exygy — APPROVED** [review]
